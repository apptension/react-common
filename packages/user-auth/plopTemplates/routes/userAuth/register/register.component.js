import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form/immutable';
import { FormattedMessage } from 'react-intl';
import { Link } from 'react-router-dom';

import {
  Container,
  Header,
  Form,
  ButtonGroup,
} from '../../../components/forms/form.styles';
import messages from './register.messages';

import { TextField } from '../../../components/forms/textField';
import { EMAIL_FIELD, PASSWORD_FIELD, PASSWORD_REPEAT_FIELD } from '../../../modules/userAuth';

export class Register extends PureComponent {
  static propTypes = {
    language: PropTypes.string,
    intl: PropTypes.object.isRequired,
    register: PropTypes.func.isRequired,
    handleSubmit: PropTypes.func.isRequired,
  };

  handleSubmit = (values) => new Promise(() => {
    this.props.register(
      values.get('email'),
      values.get('password')
    );
  });

  render() {
    const { intl, handleSubmit, language } = this.props;

    return (
      <Container>
        <Header>
          <FormattedMessage {...messages.title} />
        </Header>

        <Form onSubmit={handleSubmit(this.handleSubmit)} noValidate>
          <Field
            component={TextField}
            name={EMAIL_FIELD}
            type="email"
            label={messages.emailLabel}
            messages={messages}
            intl={intl}
          />

          <Field
            component={TextField}
            name={PASSWORD_FIELD}
            type="password"
            label={messages.passwordLabel}
            messages={messages}
            intl={intl}
          />

          <Field
            component={TextField}
            name={PASSWORD_REPEAT_FIELD}
            type="password"
            label={messages.passwordRepeatLabel}
            messages={messages}
            intl={intl}
          />

          <ButtonGroup>
            <button type="submit">
              <FormattedMessage {...messages.submit} />
            </button>
            <Link to={`/${language}/login`}>
              <FormattedMessage {...messages.signIn} />
            </Link>
          </ButtonGroup>
        </Form>
      </Container>
    );
  }
}
