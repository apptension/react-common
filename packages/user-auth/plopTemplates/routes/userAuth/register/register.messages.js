import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'register.title',
    defaultMessage: 'Sign up',
  },
  passwordLabel: {
    id: 'register.fields.password.label',
    defaultMessage: 'Password',
  },
  passwordRepeatLabel: {
    id: 'register.fields.password.passwordRepeatLabel',
    defaultMessage: 'Repeat password',
  },
  passwordPresenceError: {
    id: 'register.fields.password.error.presence',
    defaultMessage: 'Password cannot be blank',
  },
  repeatPasswordEqualityError: {
    id: 'register.fields.repeatPassword.error.equality',
    defaultMessage: 'This password does not match the one entered above. Please check that you have entered it ' +
    'correctly.',
  },
  emailLabel: {
    id: 'register.fields.email.label',
    defaultMessage: 'Email',
  },
  emailPresenceError: {
    id: 'register.fields.email.error.presence',
    defaultMessage: 'Email cannot be blank',
  },
  emailFormatError: {
    id: 'register.fields.email.error.email',
    defaultMessage: 'Email is invalid',
  },
  submit: {
    id: 'register.submit',
    defaultMessage: 'Sign up',
  },
  signIn: {
    id: 'register.signIn',
    defaultMessage: 'Sign in',
  },
});
