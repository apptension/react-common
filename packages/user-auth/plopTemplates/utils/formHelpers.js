import reportError from 'report-error';
import { List } from "immutable";
import { when, is, pipe } from 'ramda';

export const getFormFieldHelperText = (intl, messages, { error, invalid, touched }) => {
  if (invalid && touched) {
    try {
      return intl.formatMessage(messages[pipe(
        when(
          is(List),
          (errors) => errors.first()
        ),
        when(
          is(Array),
          ([firstError]) => firstError
        ),
      )(error)]);
    } catch (e) {
      reportError(e);
    }
  }
  return '';
};
