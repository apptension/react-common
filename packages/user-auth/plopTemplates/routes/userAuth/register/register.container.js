import { reduxForm } from 'redux-form/immutable';
import validate from 'validate.js';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'ramda';
import { injectIntl } from 'react-intl';

import {
  UserAuthActions,
  REGISTER_FORM,
  EMAIL_FIELD,
  PASSWORD_FIELD,
  PASSWORD_REPEAT_FIELD,
} from '../../../modules/userAuth';
import { Register } from './register.component';
import { selectLocalesLanguage } from '../../../modules/locales/locales.selectors';

const mapStateToProps = createStructuredSelector({
  language: selectLocalesLanguage,
});

export const mapDispatchToProps = (dispatch) => bindActionCreators({
  register: UserAuthActions.register,
}, dispatch);

export default compose(
  injectIntl,
  connect(mapStateToProps, mapDispatchToProps),
  reduxForm({
    form: REGISTER_FORM,
    validate: (values) => validate(values.toJS(), {
      [EMAIL_FIELD]: {
        presence: {
          message: '^emailPresenceError',
        },
        email: {
          message: '^emailFormatError',
        },
      },
      [PASSWORD_FIELD]: {
        presence: {
          message: '^passwordPresenceError',
        },
      },
      [PASSWORD_REPEAT_FIELD]: {
        presence: {
          message: '^passwordPresenceError',
        },
        equality: {
          attribute: PASSWORD_FIELD,
          message: '^repeatPasswordEqualityError',
        },
      },
    }),
  }),
)(Register);
