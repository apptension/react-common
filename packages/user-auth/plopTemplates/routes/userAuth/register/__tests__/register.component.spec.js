import React from 'react';
import { shallow } from 'enzyme';
import { spy } from 'sinon';
import { expect } from 'chai';
import { Map } from 'immutable';
import { identity } from 'ramda';

import { intlMock } from '../../../../utils/testUtils';
import { Register } from '../register.component';
import { Form } from '../../../../components/forms/form.styles';

describe('<Register />', () => {
  const defaultProps = {
    intl: intlMock(),
    handleSubmit: identity,
    register: identity,
    language: 'en',
  };

  const component = (props) => shallow(
    <Register {...defaultProps} {...props} />
  );

  it('should render itself', () => {
    const wrapper = component();
    global.expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should call register prop when handling submit', () => {
    const email = 'email@test.com';
    const password = 'passw0rd';

    const register = spy();
    const handleSubmit = spy();
    const wrapper = component({ register, handleSubmit });
    wrapper.find(Form).simulate('submit');

    const [fn] = handleSubmit.getCall(0).args;

    fn(Map({ email, password }));

    expect(register).to.have.been.calledWith(email, password);
  });
});
