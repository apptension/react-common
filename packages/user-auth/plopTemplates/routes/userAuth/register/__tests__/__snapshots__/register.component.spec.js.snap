// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> should render itself 1`] = `
.c0 {
  width: 100%;
  max-width: 600px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: auto;
}

<div
  className="c0"
>
  <styled.h1>
    <FormattedMessage
      defaultMessage="Sign up"
      id="register.title"
      values={Object {}}
    />
  </styled.h1>
  <styled.form
    noValidate={true}
    onSubmit={[Function]}
  >
    <Field
      component={[Function]}
      intl={
        Object {
          "formatMessage": [Function],
        }
      }
      label={
        Object {
          "defaultMessage": "Email",
          "id": "register.fields.email.label",
        }
      }
      messages={
        Object {
          "emailFormatError": Object {
            "defaultMessage": "Email is invalid",
            "id": "register.fields.email.error.email",
          },
          "emailLabel": Object {
            "defaultMessage": "Email",
            "id": "register.fields.email.label",
          },
          "emailPresenceError": Object {
            "defaultMessage": "Email cannot be blank",
            "id": "register.fields.email.error.presence",
          },
          "passwordLabel": Object {
            "defaultMessage": "Password",
            "id": "register.fields.password.label",
          },
          "passwordPresenceError": Object {
            "defaultMessage": "Password cannot be blank",
            "id": "register.fields.password.error.presence",
          },
          "passwordRepeatLabel": Object {
            "defaultMessage": "Repeat password",
            "id": "register.fields.password.passwordRepeatLabel",
          },
          "repeatPasswordEqualityError": Object {
            "defaultMessage": "This password does not match the one entered above. Please check that you have entered it correctly.",
            "id": "register.fields.repeatPassword.error.equality",
          },
          "signIn": Object {
            "defaultMessage": "Sign in",
            "id": "register.signIn",
          },
          "submit": Object {
            "defaultMessage": "Sign up",
            "id": "register.submit",
          },
          "title": Object {
            "defaultMessage": "Sign up",
            "id": "register.title",
          },
        }
      }
      name="email"
      type="email"
    />
    <Field
      component={[Function]}
      intl={
        Object {
          "formatMessage": [Function],
        }
      }
      label={
        Object {
          "defaultMessage": "Password",
          "id": "register.fields.password.label",
        }
      }
      messages={
        Object {
          "emailFormatError": Object {
            "defaultMessage": "Email is invalid",
            "id": "register.fields.email.error.email",
          },
          "emailLabel": Object {
            "defaultMessage": "Email",
            "id": "register.fields.email.label",
          },
          "emailPresenceError": Object {
            "defaultMessage": "Email cannot be blank",
            "id": "register.fields.email.error.presence",
          },
          "passwordLabel": Object {
            "defaultMessage": "Password",
            "id": "register.fields.password.label",
          },
          "passwordPresenceError": Object {
            "defaultMessage": "Password cannot be blank",
            "id": "register.fields.password.error.presence",
          },
          "passwordRepeatLabel": Object {
            "defaultMessage": "Repeat password",
            "id": "register.fields.password.passwordRepeatLabel",
          },
          "repeatPasswordEqualityError": Object {
            "defaultMessage": "This password does not match the one entered above. Please check that you have entered it correctly.",
            "id": "register.fields.repeatPassword.error.equality",
          },
          "signIn": Object {
            "defaultMessage": "Sign in",
            "id": "register.signIn",
          },
          "submit": Object {
            "defaultMessage": "Sign up",
            "id": "register.submit",
          },
          "title": Object {
            "defaultMessage": "Sign up",
            "id": "register.title",
          },
        }
      }
      name="password"
      type="password"
    />
    <Field
      component={[Function]}
      intl={
        Object {
          "formatMessage": [Function],
        }
      }
      label={
        Object {
          "defaultMessage": "Repeat password",
          "id": "register.fields.password.passwordRepeatLabel",
        }
      }
      messages={
        Object {
          "emailFormatError": Object {
            "defaultMessage": "Email is invalid",
            "id": "register.fields.email.error.email",
          },
          "emailLabel": Object {
            "defaultMessage": "Email",
            "id": "register.fields.email.label",
          },
          "emailPresenceError": Object {
            "defaultMessage": "Email cannot be blank",
            "id": "register.fields.email.error.presence",
          },
          "passwordLabel": Object {
            "defaultMessage": "Password",
            "id": "register.fields.password.label",
          },
          "passwordPresenceError": Object {
            "defaultMessage": "Password cannot be blank",
            "id": "register.fields.password.error.presence",
          },
          "passwordRepeatLabel": Object {
            "defaultMessage": "Repeat password",
            "id": "register.fields.password.passwordRepeatLabel",
          },
          "repeatPasswordEqualityError": Object {
            "defaultMessage": "This password does not match the one entered above. Please check that you have entered it correctly.",
            "id": "register.fields.repeatPassword.error.equality",
          },
          "signIn": Object {
            "defaultMessage": "Sign in",
            "id": "register.signIn",
          },
          "submit": Object {
            "defaultMessage": "Sign up",
            "id": "register.submit",
          },
          "title": Object {
            "defaultMessage": "Sign up",
            "id": "register.title",
          },
        }
      }
      name="passwordRepeat"
      type="password"
    />
    <styled.div>
      <button
        type="submit"
      >
        <FormattedMessage
          defaultMessage="Sign up"
          id="register.submit"
          values={Object {}}
        />
      </button>
      <Link
        replace={false}
        to="/en/login"
      >
        <FormattedMessage
          defaultMessage="Sign in"
          id="register.signIn"
          values={Object {}}
        />
      </Link>
    </styled.div>
  </styled.form>
</div>
`;
